File: frontend/apps/admin/src/rpc/service/admin_login_log.ts
import type { AdminLoginLogService } from '#/rpc/api/admin/service/v1/i_admin_login_log.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';
import type {
  AdminLoginLog,
  GetAdminLoginLogRequest,
  ListAdminLoginLogResponse,
} from '#/rpc/api/system/service/v1/admin_login_log.pb';

import { requestClient } from '#/rpc/request';

/** 后台登录日志管理服务 */
class AdminLoginLogServiceImpl implements AdminLoginLogService {
  async GetAdminLoginLog(
    request: GetAdminLoginLogRequest,
  ): Promise<AdminLoginLog> {
    return await requestClient.get<AdminLoginLog>(
      `/admin_login_logs/${request.id}`,
    );
  }

  async ListAdminLoginLog(
    request: PagingRequest,
  ): Promise<ListAdminLoginLogResponse> {
    return await requestClient.get<ListAdminLoginLogResponse>(
      '/admin_login_logs',
      {
        params: request,
      },
    );
  }
}

export const defAdminLoginLogService = new AdminLoginLogServiceImpl();

File: backend/app/admin/service/internal/service/admin_login_log.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"google.golang.org/protobuf/types/known/emptypb"

	pagination "github.com/tx7do/kratos-bootstrap/api/gen/go/pagination/v1"

	"kratos-admin/app/admin/service/internal/data"

	adminV1 "kratos-admin/api/gen/go/admin/service/v1"
)

type AdminLoginLogService struct {
	adminV1.AdminLoginLogServiceHTTPServer

	log *log.Helper

	repo *data.AdminLoginLogRepo
}

func NewAdminLoginLogService(logger log.Logger, repo *data.AdminLoginLogRepo) *AdminLoginLogService {
	l := log.NewHelper(log.With(logger, "module", "admin-login-log/service/admin-service"))
	return &AdminLoginLogService{
		log:  l,
		repo: repo,
	}
}

func (s *AdminLoginLogService) List(ctx context.Context, req *pagination.PagingRequest) (*adminV1.ListAdminLoginLogResponse, error) {
	return s.repo.List(ctx, req)
}

func (s *AdminLoginLogService) Get(ctx context.Context, req *adminV1.GetAdminLoginLogRequest) (*adminV1.AdminLoginLog, error) {
	return s.repo.Get(ctx, req)
}

func (s *AdminLoginLogService) Create(ctx context.Context, req *adminV1.CreateAdminLoginLogRequest) (*emptypb.Empty, error) {
	if req == nil || req.Data == nil {
		return nil, adminV1.ErrorBadRequest("invalid parameter")
	}

	if err := s.repo.Create(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}
